File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/view/video/AudioConfigImpl.kt
package dev.hebei.camerax_android.view.video

import androidx.camera.view.video.AudioConfig
import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiAudioConfigApi

class AudioConfigImpl(impl: CameraXImpl) : PigeonApiAudioConfigApi(impl) {
    override fun create(enableAudio: Boolean): AudioConfig {
        return AudioConfig.create(enableAudio)
    }

    override fun audioDisabled(): AudioConfig {
        return AudioConfig.AUDIO_DISABLED
    }

    override fun getAudioEnabled(pigeon_instance: AudioConfig): Boolean {
        return pigeon_instance.audioEnabled
    }
}

File: camerax_ios/ios/Classes/view/video/AudioConfigImpl.swift
//
//  AudioConfigImpl.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2025/5/19.
//

import Foundation

class AudioConfigImpl: PigeonApiDelegateAudioConfigApi {
    func create(pigeonApi: PigeonApiAudioConfigApi, enableAudio: Bool) throws -> AudioConfig {
        fatalError()
    }
    
    func audioDisabled(pigeonApi: PigeonApiAudioConfigApi) throws -> AudioConfig {
        fatalError()
    }
    
    func getAudioEnabled(pigeonApi: PigeonApiAudioConfigApi, pigeonInstance: AudioConfig) throws -> Bool {
        fatalError()
    }
}
